### ISSUE TITLE
mintty leaks kitty query

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#2245]
When we run in MinTTY on MSYS2, we have our Kitty graphics query bounced back out at us. MinTTY probably needs be taught to consume DCS or whatever that one is (OSC?). Send them a PR.

**Comment by dankamongmen** [comment]
from the MinTTY README:

> If you consider to suggest a patch or contribute to mintty otherwise, discuss your proposal in an issue first, or on the Cygwin mailing list, or with the maintainer.  **Repository policy: No unsolicited pull requests!**

fair enough =] hey @mintty, i send (along with a bunch of other stuff) the following query at startup:

`\e_Gi=1,a=q;\e\\`

to test for support of the Kitty graphics protocol. on MinTTY as shipped with MSYS2, this is not consumed, and instead echoed back onto standard output. it would be nice if MinTTY consumed this escape, instead. any chance of that happening?

**Comment by dankamongmen** [comment]
hrmmm, i confirm that a direct echo of this APC is being properly consumed. interesting. assuming this to be my bug for now, thanks @mintty.

**Comment by dankamongmen** [comment]
hrmmm, and yet it absolutely doesn't get consumed when we're writing it out of one of our programs. alright, i guess we strip until we find the phase change.

**Comment by dankamongmen** [comment]
hrmmmmmm...does MinTTY possibly reply to a series of queries out-of-order sometimes? well, still, this isn't a *reply* from mintty...hrmmm...

**Comment by dankamongmen** [comment]
hrmmmmmmmm....since i'm using ConPTY here and explicitly acquiring the ConsoleHandles etc., is it possible my queries are being answered from something other than MinTTY somehow? whereas normal programs writing to stdout aren't? that would explain very much; i think this `\e[1;0c` might also be what Microsoft Terminal replies to us with. HRMMM.

**Comment by dankamongmen** [comment]
well, if so, i ought be able to determine that pretty easily with some test code

**Comment by dankamongmen** [comment]
hrmm. when i cat that file with all the queries, i don't see a cursor location, despite `\e[6n` starting it. yet in my read data, i do see one. mystery upon mystery....

**Comment by dankamongmen** [comment]
cool, i will look into doing so, thanks! do cygwin packages naturally flow into msys without further effort on my part?

**Comment by j4james** [comment]
> if i'm understanding you correctly, running behind ConPTY as a native windows program (that is indeed what i'm building) cannot get queries through to MinTTY.

I did try to explain this to you before (https://github.com/dankamongmen/notcurses/issues/2105#issuecomment-906574179). If you're running a terminal over conpty, you're essentially running the Windows console, and most likely a version that is very out of date. If you want to test conpty, you're best off using the Windows Terminal, because that will at least have bundled the latest code.

> sounds like we want to make a patch against ConPTY to consume APCs.

This was fixed a year ago (https://github.com/microsoft/terminal/commit/f91b53d5fdc5b20387e297357668f5f14a795d4c).

**Comment by dankamongmen** [comment]
> I did try to explain this to you before ([#2105 (comment)](https://github.com/dankamongmen/notcurses/issues/2105#issuecomment-906574179)). If you're running a terminal over conpty, you're essentially running the Windows console, and most likely a version that is very out of date. If you want to test conpty, you're best off using the Windows Terminal, because that will at least have bundled the latest code.

yep, i think everything's clear now (i'd missed your comment of aug 26th).

windows terminal has always been, and remains, the primary target for windows. i would just like things to work well on MSYS+MinTTY+ConPTY as well, especially since that's currently the recommended environment in which to build our windows executables.

**Comment by j4james** [comment]
> Why would that be? Isn't conpty provided by Windows as a library? So it should be the same for all applications as I understand.

As I understand it, the conpty library just connects you to a conhost "server". By default this will be the version that is included in Windows (essentially what you get when you run something like cmd.exe). But Windows Terminal bundles its own version of conhost (called OpenConsole), which is built from the latest source.

Technically MinTTY could do the same thing - I'm fairly sure there are other third party terminals doing that - although it's not officially supported at this point.

**Comment by j4james** [comment]
I've just been doing some digging, and it looks like WezTerm is one of the terminals that does this (see https://github.com/wez/wezterm/commit/61e3545c68e9e25702b180ae4604461e9d95102d). Also, there's a bit more info on the subject here: https://github.com/microsoft/terminal/issues/1130.
